internal/pkg/runtime/engine/singularity/process_linux.go

```diff
@@ -567,3 +573,25 @@ func preStartProcess(e *EngineOperations) error {
 
 	return nil
 }
+
+func (e *EngineOperations) getIP() (string, error) {
+	if e.EngineConfig.Network == nil {
+		return "", errors.New("not in net ns")
+	}
+
+	net := e.EngineConfig.GetNetwork()
```

Comment:
`Network` can take a list of network separated by commas

```suggestion
	net := strings.Split(e.EngineConfig.GetNetwork(), ",")
```